### PR TITLE
[MBL-16849][Student] - Handle API flakiness of Notifications and Todo E2E test cases

### PR DESCRIPTION
**Pull request by kdeakinstructure**
Workaround to handle API flakiness in ToDoE2E and NotificationsE2E tests.

refs: MBL-16849
affects: Student
release note: none

### CODE REVIEW

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

File: apps/student/src/androidTest/java/com/instructure/student/ui/pages/TodoPage.kt [hunk 1]
```diff
@@ -45,6 +46,20 @@ class TodoPage: BasePage(R.id.todoPage) {
         assertTextDisplayedInRecyclerView(assignment.name)
     }
 
+    fun assertAssignmentDisplayedWithRetries(assignment: AssignmentApiModel, retryAttempt: Int) {
+        var remainingRetryAttempt = retryAttempt
+        while (remainingRetryAttempt > 0) {
```
Comment (hermannakos) [inline apps/student/src/androidTest/java/com/instructure/student/ui/pages/TodoPage.kt:51]: You can use `repeat` for this.

**Review by hermannakos — CHANGES_REQUESTED** [review]

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

File: apps/student/src/androidTest/java/com/instructure/student/ui/pages/TodoPage.kt (hunk 1 above)
Comment (kdeakinstructure) [reply apps/student/src/androidTest/java/com/instructure/student/ui/pages/TodoPage.kt]: Thanks

**Review by hermannakos — APPROVED** [review]

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Pull request merged**